assets/src/dashboard/components/typeahead/index.js

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useRef } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { KEYS } from '../../constants';
+import { Close as CloseIcon, Search as SearchIcon } from '../../icons';
+import useFocusOut from '../../utils/useFocusOut';
+import TypeaheadOptions from './typeaheadOptions';
+import useTypeahead from './useTypeahead';
+import {
+  SearchContainer,
+  InputContainer,
+  ControlVisibilityContainer,
+  StyledInput,
+  SearchButton,
+  ClearInputButton,
+} from './components';
+
+const Typeahead = ({
+  inputId,
+  items,
+  className,
+  disabled,
+  onChange,
+  placeholder,
+  value = '',
+  ariaLabel,
+  isFiltering,
+  ...rest
+}) => {
+  const {
+    isMenuOpen,
+    isInputExpanded,
+    showMenu,
+    menuFocused,
+    inputValue,
+    selectedValueIndex,
+  } = useTypeahead({ items, value });
+
+  const searchRef = useRef();
+  const inputRef = useRef();
+  const typeaheadOptionsRef = useRef();
+
+  const focusInput = useCallback(() => {
+    inputRef.current.focus();
+  }, [inputRef]);
+
+  const handleFocusOut = useCallback(() => {
+    showMenu.set(false);
+    menuFocused.set(false);
+  }, [showMenu, menuFocused]);
+
+  useFocusOut(searchRef, handleFocusOut);
+
+  const handleInputChange = useCallback(
+    (item) => {
+      if (!showMenu.value) {
+        showMenu.set(true);
+      }
+      inputValue.set(item.label);
+      onChange(item.value.trim());
+    },
+    [onChange, inputValue, showMenu]
+  );
+
+  const handleMenuItemSelect = useCallback(
+    (item) => {
+      if (!item.value) {
+        return;
+      }
+      handleInputChange(item);
+      showMenu.set(false);
+    },
+    [handleInputChange, showMenu]
+  );
+
+  const handleClearInputButtonFocus = useCallback(() => {
+    showMenu.set(false);
+  }, [showMenu]);
+
+  const handleInputClear = useCallback(() => {
+    handleInputChange({ label: '', value: '' });
+    showMenu.set(false);
+    menuFocused.set(false);
+  }, [handleInputChange, menuFocused, showMenu]);
+
+  const handleKeyDown = useCallback(
+    (event) => {
+      if (event.key === KEYS.DOWN) {
+        event.preventDefault();
+        typeaheadOptionsRef.current?.children[selectedValueIndex.value].focus();
```

Comment:
I think this is where it is happening. Either the ref doesn't exit or the children haven't been populated. 

Alternatively this might not even be called when the text box has been editing and the key down is pressed.